x-pack/legacy/plugins/lens/public/app_plugin/url_helper.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { DashboardConstants } from '../../../../../../src/legacy/core_plugins/kibana/public/dashboard';
+
+const EMPTY_DASHBOARD_PATTERN = /(.*#\/dashboard\?)(.*)/;
+const DASHBOARD_WITH_ID_PATTERN = /(.*#\/dashboard\/.*\?)(.*)/;
+
+/** *
+ * Returns base path from dashboard url
+ * eg.
+ * input: http://localhost:5601/lib/app/kibana#/dashboard?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_a=(description:'',filters:!(),fullScreenMode:!f,options:(hidePanelTitles:!f,useMargins:!t),panels:!(),query:(language:kuery,query:''),timeRestore:!f,title:'',viewMode:edit)
+ * output: http://localhost:5601/lib/app/kibana#
+ * @param url
+ */
+export function getKibanaBasePathFromDashboardUrl(url: string | undefined): string | null {
```

Comment:
Rather than parsing this from the URL, shouldn't we just use Kibana's base path utilities? (e.g. `core.http.basePath` or something like that)